server/lib/ga-collector.js

```diff
@@ -103,6 +106,30 @@ GACollector.prototype = {
         });
       }
     });
+  },
+  /**
+   * Calculate the queue time of the event.
+   * See https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters?hl=en#qt
+   *
+   * @param {Number} startTime
+   *  Time when metrics object was create on the front-end
+   * @param {Number} offset
+   *  Offset of the event since metrics start
+   *
+   * @returns {Number}
+   *  The time delta (in milliseconds) between when the hit being reported occurred and the time the hit was sent.
+   *
+   * @private
+   */
+  _calculateQueueTime: function(startTime, offset) {
+    if (! startTime || ! offset) {
+      return 0;
+    }
+
+    var dispatchTime = Date.now();
+    var offsetTime = startTime + offset;
```

Comment:
There is a classic distributed systems problem with this calculation - clock skew between the server and client. If the server's time is ahead of the `startTime + offset` as reported by the client, the queue time will be negative, and [Google will ignore the event](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters?hl=en#qt).

To avoid clock skew, we either have to have a way to synchronize the clock for the purpose of calculations, or depend only on one clock. I would depend on the client's clock if at all possible. Instead of having the front end report only a `startTime`, we could have it also report a `dispatchTime`. `dispatchTime` would be used on line 129 as the right hand value.